src/test/java/core/basesyntax/strategy/impl/PurchaseOperationTest.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.exception.InvalidOperatioExeption;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationTest {
+    private static Storage storageImpl;
+    private static PurchaseOperation purchaseOperation;
+
+    @BeforeAll
+    static void setUp() {
+        storageImpl = new StorageImpl();
```

Comment:
your solution is quite inconsistent. Sometimes you reference Storage directly and sometimes you create a StorageImpl for these purposes